cs/src/core/Epochs/LightEpoch.cs

```diff
@@ -132,23 +121,17 @@ void Uninitialize()
         /// <returns>Result of the check</returns>
         public bool IsProtected()
         {
-            return (kInvalidIndex != threadEntryIndex);
+            return kInvalidIndex != threadEntryIndex.Value;
```

Comment:
>threadEntryIndex.Value; [](start = 36, length = 23)

This can be called from an async callback thread that has not initialized itself, so we should check to see if it's initialized before derefing the value